consensus/module.go

```diff
@@ -30,6 +29,7 @@ type consensusModule struct {
 	privateKey cryptoPocket.Ed25519PrivateKey
 	consCfg    *config.ConsensusConfig
 
+
```

Comment:
1. Yes, I played with it and slightly modified with it, but I thought any addition of core library functionality still needs to go through a review, because just because it's a library doesn't mean it doesn't have problems.
2. Core logic. Play with it if you want, I have fairly extensively.
3. I put it in there because I thought we would modify it more extensively, and I'm concerned about not having a versioned instance of that.

Reply:
4. Yes
5. Yes
6. No, I thought we wanted to move away from larger PRs and this was the smallest I was willing to make a PR. Having primitives to actually do the thing you're trying to accomplish seems like a prerequisite.